### DISCUSSION TITLE
DSA公私钥格式问题

### DISCUSSION THREAD

**Discussion by songjinlian** [Leon406/ToolsFx#337] [Q&A] [answered]
导入OpenSSL生成的密钥对，签名时提示私钥不存在。
![图片](https://github.com/user-attachments/assets/8e706c8e-5aaf-476d-8f27-ae295b766ccc)

**Comment by songjinlian** [comment]
[Uploading 密钥对.zip…]()

**Comment by Leon406** [comment, marked as answer]
附件没有上传成功

可以先把秘钥头尾删除看看

最新github acion包看看有没有解决

**Reply by songjinlian** [reply]
测试过了，可以了。
DSA有两种签名模式：
1、RFC 6979方式，该方式签名时未引入随机数，签名值不变；
2、FIPS-186-3方式，该方式签名时引入随机数，签名值每次都变。

**Reply by songjinlian** [reply]
【非对称加密】模块，是否可以加入ECDSA、DSA的密钥对生成功能？

**Reply by Leon406** [reply]
这个后面再看吧，预计在签名和验签里加
